src/main/java/io/cdap/plugin/hubspot/common/HubspotHelper.java

```diff
@@ -0,0 +1,288 @@
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import org.apache.http.Header;
+import org.apache.http.HttpEntity;
+import org.apache.http.StatusLine;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.message.BasicHeader;
+import org.apache.http.util.EntityUtils;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Helper class to incorporate Hubspot api interaction
+ */
+public class HubspotHelper {
+
+  public HubspotPage getHupspotPage(BaseHubspotConfig baseHubspotConfig, String offset) throws IOException {
+    HttpGet request = getRequest(baseHubspotConfig, offset);
+
+    CloseableHttpResponse response = downloadPage(baseHubspotConfig, request);
+    HttpEntity entity = response.getEntity();
+    String result;
+    if (entity != null) {
+      result = EntityUtils.toString(entity);
+      return parseJson(baseHubspotConfig, result);
+    }
+    return null;
+  }
+
+  private CloseableHttpResponse downloadPage(BaseHubspotConfig baseHubspotConfig, HttpGet request) throws IOException {
+    HttpClientBuilder httpClientBuilder = HttpClientBuilder.create();
+    ArrayList<Header> clientHeaders = new ArrayList<>();
+    String accessToken = baseHubspotConfig.apiKey;
+    clientHeaders.add(new BasicHeader("Authorization", "Bearer " + accessToken));
+    httpClientBuilder.setDefaultHeaders(clientHeaders);
+    CloseableHttpClient client = httpClientBuilder.build();
+
+    int maxTries = 3;
+    int count = 0;
+    while (true) {
+      CloseableHttpResponse response = client.execute(request);
+      StatusLine statusLine = response.getStatusLine();
+      if (statusLine.getStatusCode() != 200) {
+        if (500 > statusLine.getStatusCode() &&
+          statusLine.getStatusCode() >= 400) {
+          throw new IOException(statusLine.getReasonPhrase());
+        }
+        if (++count == maxTries) {
+          throw new IOException(statusLine.getReasonPhrase());
+        }
+      }
+      return response;
+    }
+  }
+
+  private HttpGet getRequest(BaseHubspotConfig baseHubspotConfig, String offset) {
+    URI uri = null;
+    try {
+      URIBuilder b = new URIBuilder(getEndpoint(baseHubspotConfig));
+      if (baseHubspotConfig.startDate != null) {
+        b.addParameter("start", baseHubspotConfig.startDate);
+      }
+      if (baseHubspotConfig.endDate != null) {
+        b.addParameter("end", baseHubspotConfig.endDate);
+      }
+      for (String filter : baseHubspotConfig.getFilters()) {
+        b.addParameter("f", filter);
+      }
+      if (getLimitPropertyName(baseHubspotConfig) != null) {
+        b.addParameter(getLimitPropertyName(baseHubspotConfig), "100");
```

Comment:
No magic numbers should be in code. `private static final` with a good descriptive name. and a comment what is this and why we use this value.